include/nbl/macros.h

```diff
@@ -81,7 +81,7 @@
 
 //! Workarounds for compiler specific bugs
 // MSVC 2019 is a special snowflake
-#if defined(_MSC_VER) && _MSC_VER>=1920
+#if defined(_MSC_VER) && !defined(__clang__) && _MSC_VER>=1920

```

Comment:
can we just ban old MSC_VER and old GCC ? and not do weird macro tricks ( or less)

Reply:
we can ban old compilers at header level, there are a few headers checking for compiler version and doing weird tricks so imo we should do it in another PR and delegate @YasInvolved